### DISCUSSION TITLE
Whisper's output for Hindi audio leads to a transcript in Arabic or Persian script

### DISCUSSION THREAD

**Discussion by abhilashgupta** [openai/whisper#1662] [Q&A]
I tried transcribing a Hindi (non standard dialect) audio file.
The API call went through fine and returned the transcription in the Devnagiri script.
However, when I run whisper transcribe on it locally, it recognises that the audio is in Hindi, but the transcription is in either Arabic or Persian script (or maybe Urdu, not sure).
Same when I force it to transcribe in Hindi, Nepali or Marathi language (almost same script as Hindi, one extra character).

Writing here because I couldn't see an option to raise an issue.

Steps to reproduce:
I cannot attach a `.wav` file here, so here is the link to the video -> https://www.youtube.com/watch?v=x2CnZZFf8Jg
To use API, call the API and get back response in correct script.
To run locally, just run `whisper downloaded_audio.wav --language hi --model base` or `whisper downloaded_audio.wav --model base`

There's a conversation about how Whisper's auto detection detects `Hindi` audio to be `Urdu` (which makes sense given their similarity) at https://github.com/openai/whisper/discussions/118 but the API gives the response in the correct script and the toolkit outputs it in the wrong script even if I explicitly mention the language

**Comment by 7gxycn08** [comment]
Don't bother using Whisper the accuracy rate is pretty low compared to letting GPT3.5 do the translation.

**Reply by abhilashgupta** [reply]
I am not translating here.
I have an audio file from which I want to extract the transcription of the audio in the same language, which is a very different problem than translation, no?

**Comment by kennethleungty** [comment]
@abhilashgupta Did you come across any solution/new insights for this?

**Reply by abhilashgupta** [reply]
Yes,
the issue was observed in the `base` model, however it isn't the case with the `large` (large-v2 or large-v3) models. So, I started using the large models wherever possible.
However, occasionally, it still detects the language `Hindi` as `Urdu`, when set to auto-detect, given that they both are close to each other, and in that case, it still returns the `Urdu` script transcriptions. For that, always set the language manually to Hindi where possible.

**Reply by Aliraza1010a** [reply]
Bhai me bhi ye issue face kar raha hu 
Please batao kaise isse fix kare

**Comment by nairajay2k** [comment]
I am facing the same issue. When I call model.transcribe('filename', language='hi') the returned text is in urdu/persian.

**Reply by gaurav241102** [reply]
did you come up with any solution for this?

**Comment by firofame** [comment]
same issue with Malayalam audio

```
model = whisper.load_model('large-v3')
result = model.transcribe("audio.mp3")
print(result["language"]) // correct language detected i.e "ml"
print(result["text"]) // text is in a different language
```
